base/missing.jl

```diff
@@ -345,10 +354,19 @@ mapreduce_impl(f, op, A::SkipMissing, ifirst::Integer, ilast::Integer) =
         i == typemax(typeof(i)) && return Some(op(f(a1), f(a2)))
         i += 1
         v = op(f(a1), f(a2))
-        @simd for i = i:ilast
-            @inbounds ai = A[i]
-            if ai !== missing
-                v = op(v, f(ai))
+        # We need to make sure `fskip` is stable.
+        noop = _fast_noop(op, _return_type(f, Tuple{nonmissingtype(eltype(A))}), v)
+        if isnothing(noop)
```

Comment:
Do both branches provide bit-identical results? Otherwise, relying on the inference result like this is not an optimization and it introduces unpredictable behavior.

Reply:
1. For bit-identity, I test it locally and there seems no problem. (`Float16` is not a `HWReal`)
2. `_return_type(f, Tuple{nonmissingtype(eltype(A))})` was used to make sure `fskip(x) = ismissing(x) ? noop : f(x)` is stable. (In this sence, I agree that we'd better fix this at LLVM level)

Reply:
The following seems also work for the original purpose:
```julia
noop = _fast_noop(op, typeof(v), v) # 
@simd for i = i:ilast
     @inbounds ai = A[i]
     v = op(v, ismissing(ai) ? oftype(v, noop) : f(ai))
end
```
But for inputs like `Union{Int,Int32,Missing}` there're about 5%~10% speed regression. Not sure is it Ok.